modules/social_features/social_emoji/assets/js/social_emoji_picker.js

```diff
@@ -0,0 +1,377 @@
+      this.button.classList.add('with-editor');
+    }
+
+    this.input.after(this.button);
+  };
+
+  /**
+   * Checks if element has been focused by a user.
+   *
+   * By default, caret position is set to the start of the text, which creates
+   * a weird bug when the picker is being used without touching the input
+   * beforehand.
+   *
+   * This makes sense for plain inputs only, as CKEditor tracks this by itself.
+   */
+  Drupal.emojiPickerTrigger.prototype.bindInputFocus = function() {
+    this.input.addEventListener('focus', e => {
+      this.inputHasBeenFocused = true;
+    }, {once: true});
+  };
+
+  /**
+   * Opens the emoji picker upon click on the trigger button.
+   */
+  Drupal.emojiPickerTrigger.prototype.bindButtonClick = function() {
+    this.button.addEventListener('click', (e) => {
+      e.preventDefault();
+      this.emojiPicker.showPicker(this);
+    }, true);
+  };
+
+  /**
+   * Moves caret (cursor position) to the end of the text.
+   */
+  Drupal.emojiPickerTrigger.prototype.moveCaretToTheEnd = function() {
+    if (this.hasEditor()) {
+      const range = CKEDITOR.instances[this.input.id].createRange();
+      range.moveToElementEditEnd(range.root);
+      CKEDITOR.instances[this.input.id].getSelection().selectRanges([range]);
+    }
+    else {
+      const newCursorPosition = this.input.value.length;
+      this.input.setSelectionRange(newCursorPosition, newCursorPosition);
+    }
+  };
+
+  /**
+   * Inserts the given emojhi character to the corresponding input DOM element.
+   *
+   * This method ensures that empji is being inserted into the right input
+   * DOM element, and that caret (cursor) position is being preserved.
+   *
+   * @param {string} emoji
+   *   The emoji character to insert (unicode).
+   */
+  Drupal.emojiPickerTrigger.prototype.insertEmoji = function(emoji) {
+    const input = this.input;
+
+    // Move caret (cursor position) to the end if has not been focused by a
+    // user.
+    if (!this.hasBeenFocused()) {
+      this.moveCaretToTheEnd();
+    }
+
+    if (this.hasEditor()) {
+      CKEDITOR.instances[input.id].insertText(emoji);
+    }
+    else {
+      const currentValue = this.input.value,
+            newCursorPosition = input.selectionStart + emoji.length;
+      input.value = currentValue.slice(0, input.selectionStart) + emoji + currentValue.slice(input.selectionEnd);
+      input.setSelectionRange(newCursorPosition, newCursorPosition);
+
+      // Mentions.js creates a hidden element and (unfortunately) does not
+      // update it when text is being changed programatically. Fix this by doing
+      // so manually.
+      const parent = input.parentElement;
+      if (parent.classList.contains('mentions-input')) {
+        parent.querySelector('input[type="hidden"]').value = input.value;
+      }
+    }
+  };
+
+  /**
+   * Returns focus to the corresponding input DOM element.
+   */
+  Drupal.emojiPickerTrigger.prototype.returnFocus = function() {
```

Comment:
It seems this doesn't work on iOS (I doubt it's browser related but can't rule it out), at least after testing it out on an iPhone 13 the focus isn't returned after the picker is closed. Is that something you can still take a look at?
On my Android it seems to work fine.

Reply:
Focus returning won't work on iOS afaik, as _I believe_ iOS requires user interaction in order to set focus.

Reply:
Check, thanks, i'll leave the comment open so people are aware of this.